packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -323,6 +331,69 @@ export async function uploadData<T extends UploadDataLookup, K extends keyof T>(
   );
 }
 
+export function shrinkRawData<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
+): Boolean {
+  let totalSize = Buffer.byteLength(JSON.stringify(data));
+
+  while (totalSize > UPLOAD_SIZE_MAX) {
+    let largestEntryKey = '';
+    let largestEntrySize = 0;
+    let largestRawDataEntryKey = '';
+    let largestRawDataEntrySize = 0;
+    let largestItemKey = '';
+    let largestItemSize = 0;
+
+    // Find largest Entity
+    for (const entry in data) {
+      if (JSON.stringify(data[entry]).length > largestEntrySize) {
+        largestEntrySize = JSON.stringify(data[entry]).length;
+        largestEntryKey = entry;
+      }
+    }
+
+    // Find largest _rawData entry (typically 0, but check to be certain)
+    for (const rawEntry in data[largestEntryKey]['_rawData']) {
+      if (
+        JSON.stringify(data[largestEntryKey]['_rawData'][rawEntry]).length >
+        largestRawDataEntrySize
+      ) {
+        largestRawDataEntrySize = JSON.stringify(
+          data[largestEntryKey]['_rawData'][rawEntry],
+        ).length;
+        largestRawDataEntryKey = rawEntry;
+      }
+    }
+
+    // Find largest item within rawData
```

Comment:
It'd be interesting to know the benefit of find the largest N items. If there happens to be 2-3 items that are very similar and thus similar in size, it'd be better to remove those items all at the same time. Maybe worth considering now or in the future. Hard to know how to optimize without more data examples.